components/eamxx/cime_config/SystemTests/mvkxx.py

```diff
@@ -21,13 +25,195 @@
 from CIME.case.case_setup import case_setup
 from CIME.XML.machines import Machines
 
+# TODO: all the ensure_pip is a HACK
+# TODO: not sure if eamxx codebase will play nicely with conda env on gpu machines
+# TODO: the whole thing is a bit convoluted and whackily designed
+# TODO: so just do more whacky stuff here to get around the mess
+# HACK: BEGIN
+def run_cmd(cmd, input_str=None, from_dir=None, verbose=None, dry_run=False,
+            arg_stdout=subprocess.PIPE, arg_stderr=subprocess.PIPE, env=None, combine_output=False):
+###############################################################################
+    """
+    Wrapper around subprocess to make it much more convenient to run shell commands
+
+    >>> run_cmd('ls file_i_hope_doesnt_exist')[0] != 0
+    True
+    """
+    arg_stderr = subprocess.STDOUT if combine_output else arg_stderr
+    from_dir = str(from_dir) if from_dir else from_dir
+
+    if verbose:
+        print("RUN: {}\nFROM: {}".format(cmd, os.getcwd() if from_dir is None else from_dir))
+
+    if dry_run:
+        return 0, "", ""
+
+    if input_str is not None:
+        stdin = subprocess.PIPE
+        input_str = input_str.encode('utf-8')
+    else:
+        stdin = None
+
+    proc = subprocess.Popen(cmd,
+                            shell=True,
+                            stdout=arg_stdout,
+                            stderr=arg_stderr,
+                            stdin=stdin,
+                            cwd=from_dir,
+                            env=env)
+
+    output, errput = proc.communicate(input_str)
+    if output is not None:
+        try:
+            output = output.decode('utf-8', errors='ignore')
+            output = output.strip()
+        except AttributeError:
+            pass
+    if errput is not None:
+        try:
+            errput = errput.decode('utf-8', errors='ignore')
+            errput = errput.strip()
+        except AttributeError:
+            pass
+
+    stat = proc.wait()
+
+    return stat, output, errput
+
+###############################################################################
+def ensure_pip():
```

Comment:
This code lives near eamxx/scripts. Why not just import all this stuff from there?